# Get population and sample

In [2]:
import pandas as pd
import sqlite3
conn = sqlite3.connect('data/im.db')
cur = conn.cursor()
import seaborn as sbn
from scipy import stats
import matplotlib.pyplot as plt
%matplotlib inline

import numpy as np

In [4]:
poblation.describe()

runtime_minutes      numvotes  averagerating
count       217.000000  2.170000e+02     217.000000
mean        123.387097  4.543959e+05       7.395853
std          19.746951  2.101956e+05       0.653149
min          83.000000  2.509760e+05       4.100000
25%         108.000000  3.155840e+05       7.000000
50%         123.000000  3.955720e+05       7.500000
75%         136.000000  5.358360e+05       7.800000
max         181.000000  1.841066e+06       8.800000

In [3]:
poblation= pd.read_sql("""
SELECT mb.runtime_minutes, mr.numvotes, mr.averagerating
  FROM movie_basics mb
  JOIN movie_ratings mr USING(movie_id) 
  WHERE mb.runtime_minutes AND mb.genres IS NOT Null 
  AND mb.runtime_minutes < 210 AND mb.runtime_minutes > 30
  AND mr.numvotes > 250000
  
  
  ;
""", conn)
#ORDER BY mr.numvotes, mb.runtime_minutes DESC
#AND mr.numvotes > 400000
#  AND mr.averagerating > 6
poblation.info()
#poblation.tail()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 217 entries, 0 to 216
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   runtime_minutes  217 non-null    float64
 1   numvotes         217 non-null    int64  
 2   averagerating    217 non-null    float64
dtypes: float64(2), int64(1)
memory usage: 5.2 KB


# Population

In [5]:
p_mean=poblation["runtime_minutes"].mean()
p_std=poblation["runtime_minutes"].std()
print(p_mean,p_std)

123.38709677419355 19.746950523009474


# Sample

In [8]:
# duration between 90 and 130 minutes and more than 250000 votes
sample= pd.read_sql("""
SELECT mb.runtime_minutes, mr.numvotes, mr.averagerating
  FROM movie_basics mb
  JOIN movie_ratings mr USING(movie_id) 
  WHERE mb.runtime_minutes AND mb.genres IS NOT Null 
  AND mb.runtime_minutes < 130 AND mb.runtime_minutes > 90
  AND mr.numvotes > 250000
  
  
  ;
""", conn)
sample.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131 entries, 0 to 130
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   runtime_minutes  131 non-null    float64
 1   numvotes         131 non-null    int64  
 2   averagerating    131 non-null    float64
dtypes: float64(2), int64(1)
memory usage: 3.2 KB


In [9]:
sample.describe()

runtime_minutes       numvotes  averagerating
count       131.000000     131.000000     131.000000
mean        111.366412  413693.473282       7.303817
std          10.316683  138300.815311       0.678959
min          91.000000  252281.000000       4.100000
25%         103.000000  305005.500000       6.900000
50%         112.000000  371204.000000       7.400000
75%         120.000000  514996.500000       7.800000
max         129.000000  948394.000000       8.500000

In [11]:
s_mean=sample["runtime_minutes"].mean()
s_std=sample["runtime_minutes"].std()
print(s_mean,s_std)

111.36641221374046 10.316682611464588


# Z - score

In [12]:
z = (s_mean - p_mean)/(p_std/np.sqrt(78))
z

-5.37621296836578

In [13]:
stats.norm.cdf(z)

3.803440506093555e-08

In [14]:
print(1 - stats.norm.cdf(z))

0.9999999619655949


# Hypotesis
H0 A movie with a duration of 120 min will have more than 250 000 votes as well as a big audience 
HA A movie with a duration of 120 min doesn't have more than 250 000 votes as well as a big audience

In [ ]:
# alpha = 0.05
# we reject null hypotesis